extensions/Worker.Extensions.Tables/src/TableConverter.cs

```diff
@@ -0,0 +1,178 @@
+        {
+            content.TryGetValue(Constants.Connection, out var connection);
+            content.TryGetValue(Constants.TableName, out var tableName);
+            content.TryGetValue(Constants.PartitionKey, out var partitionKey);
+            content.TryGetValue(Constants.RowKey, out var rowKey);
+
+
+            if (string.IsNullOrEmpty(tableName?.ToString()))
+            {
+                throw new ArgumentNullException("'TableName' cannot be null or empty");
+            }
+
+            return await ToTargetTypeAsync(targetType, connection?.ToString() ?? null, tableName!.ToString(), partitionKey?.ToString() ?? null, rowKey?.ToString() ?? null);
+        }
+
+        internal Dictionary<string, object> GetBindingDataContent(ModelBindingData bindingData)
+        {
+            return bindingData?.ContentType switch
+            {
+                Constants.JsonContentType => new Dictionary<string, object>(bindingData?.Content?.ToObjectFromJson<Dictionary<string, object>>(), StringComparer.OrdinalIgnoreCase),
+                _ => throw new NotSupportedException($"Unexpected content-type. Currently only '{Constants.JsonContentType}' is supported.")
+            };
+        }
+
+        internal virtual async Task<object?> ToTargetTypeAsync(Type targetType, string? connection, string tableName, string? partitionKey, string? rowKey) => targetType switch
+        {
+            Type _ when targetType == typeof(TableClient) => GetTableClient(connection, tableName),
+            Type _ when targetType == typeof(TableEntity) => GetTableEntity(connection, tableName, partitionKey, rowKey),
+            _ => null
+        };
+
+        internal object? ToTargetTypeCollection(IEnumerable<object> tableCollection, string methodName, Type type)
+        {
+            tableCollection = tableCollection.Select(b => Convert.ChangeType(b, type));
+            MethodInfo method = typeof(TableConverter).GetMethod(methodName, BindingFlags.Static | BindingFlags.NonPublic);
+            MethodInfo genericMethod = method.MakeGenericMethod(type);
+
+            return genericMethod.Invoke(null, new[] { tableCollection.ToList() });
+        }
+
+        internal virtual TableClient GetTableClient(string? connection, string tableName)
+        {
+            var tableOptions = _tableOptions.Get(connection);
+            TableServiceClient tableServiceClient = tableOptions.CreateClient();
+            return tableServiceClient.GetTableClient(tableName);
+        }
+
+        internal virtual TableEntity GetTableEntity(string? connection, string tableName, string? partitionKey, string? rowKey)
+        {
+            var tableOptions = _tableOptions.Get(connection);
+            TableServiceClient tableServiceClient = tableOptions.CreateClient();
+            var tableClient = tableServiceClient.GetTableClient(tableName);
+            return tableClient.GetEntity<TableEntity>(partitionKey, rowKey);
```

Comment:
`GetTableEntity` and `GetTableClient` have the exact same code except for entity we do `tableClient.GetEntity<TableEntity>(partitionKey, rowKey);` - this can be consolidated into one function either using generics (or a simply if-statement checking the type) or via a 3rd function that will create and return the table client

Reply:
done!